ui/src/range_slider.tsx

```diff
@@ -54,17 +54,26 @@ export interface RangeSlider {
 
 export const XRangeSlider = ({ model }: { model: RangeSlider }) => {
   const
-    { min = 0, max = 100, step = 1, min_value, max_value = max, disabled, trigger, name, label } = model,
-    onChange = React.useCallback((_val: U, val_range?: [U, U]) => { if (val_range) wave.args[name] = val_range }, [name]),
+    { min = 0, max = 100, step = 1, min_value = min, max_value = max, disabled, trigger, name, label } = model,
+    [valRange, setValRange] = React.useState<[U, U]>([min_value, max_value]),
+    onChange = React.useCallback((_val: U, val_range?: [U, U]) => {
+      if (val_range) {
+        wave.args[name] = val_range
+        model.min_value = val_range[0]
+        model.max_value = val_range[1]
+        setValRange(val_range)
+      }
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [name]),
     onChanged = React.useCallback(() => { if (trigger) wave.push() }, [trigger])
 
   React.useEffect(() => {
-    wave.args[name] = [
-      typeof min_value == 'number' && min_value > min && min_value <= max ? min_value : min,
-      typeof max_value == 'number' && max_value > min && max_value <= max ? max_value : max,
-    ]
+    const lowerValue = typeof min_value == 'number' && min_value > min && min_value <= max ? min_value : min
+    const value = typeof max_value == 'number' && max_value > min && max_value <= max ? max_value : max
```

Comment:
Nit: naming

```suggestion
    const upperValue = typeof max_value == 'number' && max_value > min && max_value <= max ? max_value : max
```

Reply:
I've tried to be consistent with FluentUI's props `lowerValue` and `value`, however `lowerValue` and `upperValue` makes much more sense.